### PR TITLE
profile education form edit

### PR DESCRIPTION
**Pull request by muammertop**

### CODE REVIEW

File: addons/default/visiosoft/profile-module/resources/config/permissions.php [hunk 1]
```diff
@@ -6,4 +6,24 @@
...
+    'education_part' => [
+        'read',
+        'write',
+        'delete',
+    ],
+    'education_part_option' => [
+        'read',
+        'write',
+        'delete',
+    ],
```
Comment (diashalabi) [inline addons/default/visiosoft/profile-module/resources/config/permissions.php:28]: we dont have a stream for this "education_part_option", "education" is duplicated

File: addons/default/visiosoft/profile-module/resources/assets/js/education.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    $('#education').on('change', () => {
+        $.ajax({
+            url: '/ajax/setEducation',
+            data: {
+                info: 'education',
+                education: $('#education').val()
+            },beforeSend: function (){
+                $('#education_part').html('');
+            },success: function (response) {
+                $('#education_part').html('<option>'+ choose_an_option +'</option>')
```
Comment (diashalabi) [inline addons/default/visiosoft/profile-module/resources/assets/js/education.js:22]: '<option value="">'+ choose_an_option +'</option>' or else the value will be "choose an option"

File: addons/default/visiosoft/profile-module/resources/assets/js/education.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+            data: {
+                info: 'education_part',
+                education: $('#education_part').val()
+            }, beforeSend: function (){
+                $('#education_part_option').html('');
+            }, success: function (response) {
+
+            }
+        })
+    })
```
Comment (diashalabi) [inline addons/default/visiosoft/profile-module/resources/assets/js/education.js:44]: unused function

File: addons/default/visiosoft/profile-module/resources/lang/en/button.php [hunk 4]
```diff
@@ -15,4 +15,8 @@
     'personal' => 'Personal',
     'corporate' => 'Corporate',
     'export' => 'Export',
+    'new_education' => 'New Education',
+    'new_education' => 'New Education',
+    'new_education_part' => 'New Education part',
+    'new_education_part_option' => 'New Education part option',
```
Comment (diashalabi) [inline addons/default/visiosoft/profile-module/resources/lang/en/button.php:21]: unused "new_education_part_option". you can remove duplicated "new_education"

File: addons/default/visiosoft/profile-module/resources/lang/en/permission.php [hunk 5]
```diff
@@ -17,4 +17,36 @@
...
+        ],
+    ],
+    'education_part_option' => [
+        'name'   => 'Education part option',
+        'option' => [
+            'read'   => 'Can read education part option?',
+            'write'  => 'Can create/edit education part option?',
+            'delete' => 'Can delete education part option?',
+        ],
+    ],
```
Comment (diashalabi) [inline addons/default/visiosoft/profile-module/resources/lang/en/permission.php:51]: unused "education_part_option". you can remove duplicated "education"

File: addons/default/visiosoft/profile-module/resources/lang/en/section.php [hunk 6]
```diff
@@ -9,4 +9,16 @@
...
+    ],
+    'education' => [
+        'title' => 'Education',
+    ],
+    'education_part' => [
+        'title' => 'Education part',
+    ],
+    'education_part_option' => [
+        'title' => 'Education part option',
+    ],
```
Comment (diashalabi) [inline addons/default/visiosoft/profile-module/resources/lang/en/section.php:23]: unused "education_part_option". you can remove duplicated "education"

File: addons/default/visiosoft/profile-module/resources/lang/en/stream.php [hunk 7]
```diff
@@ -7,4 +7,16 @@
...
+    ],
+    'education' => [
+        'name' => 'Education',
+    ],
+    'education_part' => [
+        'name' => 'Education part',
+    ],
+    'education_part_option' => [
+        'name' => 'Education part option',
+    ],
```
Comment (diashalabi) [inline addons/default/visiosoft/profile-module/resources/lang/en/stream.php:21]: unused "education_part_option". you can remove duplicated "education"

File: addons/default/visiosoft/profile-module/resources/views/profile/detail.twig [hunk 8]
```diff
@@ -171,40 +171,15 @@
...
-                                                    {{ trans('visiosoft.module.profile::field.choose_an_option') }}
-                                                </option>
-                                                {% for education in setting_value('visiosoft.module.profile::education') %}
-                                                    <option value="{{ education }}"
-                                                            {{ profileForm.fields.education.value == education ? 'selected' }}>
-                                                        {{ education }}
-                                                    </option>
-                                                {% endfor %}
-                                            </select>
+                                            {{ profileForm.fields.education.input|raw }}
```
Comment (diashalabi) [inline addons/default/visiosoft/profile-module/resources/views/profile/detail.twig:174]: you can translate the "choose an option"

File: addons/default/visiosoft/profile-module/resources/views/profile/detail.twig [hunk 9]
```diff
@@ -171,40 +171,15 @@
...
-                                                    {{ trans('visiosoft.module.profile::field.choose_an_option') }}
-                                                </option>
-                                                {% for state_of_education in setting_value('visiosoft.module.profile::state_of_education') %}
-                                                    <option value="{{ state_of_education }}"
-                                                            {{ profileForm.fields.state_of_education.value == state_of_education ? 'selected' }}>
-                                                        {{ state_of_education }}
-                                                    </option>
-                                                {% endfor %}
-                                            </select>
+                                            {{ profileForm.fields.education_part.input|raw }}
```
Comment (diashalabi) [inline addons/default/visiosoft/profile-module/resources/views/profile/detail.twig:182]: you can translate the "choose an option"

File: addons/default/visiosoft/profile-module/src/Education/Form/EducationFormBuilder.php [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+     * @var array
+     */
+    protected $sections = [];
+
+    /**
+     * The form assets.
+     *
+     * @var array
+     */
+    protected $assets = [];
```
Comment (diashalabi) [inline addons/default/visiosoft/profile-module/src/Education/Form/EducationFormBuilder.php:64]: you can remove unused properties

File: addons/default/visiosoft/profile-module/src/Education/Table/EducationTableBuilder.php [hunk 11]
```diff
@@ -0,0 +1,61 @@
...
+     * @var array
+     */
+    protected $options = [];
+
+    /**
+     * The table assets.
+     *
+     * @var array
+     */
+    protected $assets = [];
```
Comment (diashalabi) [inline addons/default/visiosoft/profile-module/src/Education/Table/EducationTableBuilder.php:59]: you can remove unused properties

File: addons/default/visiosoft/profile-module/src/EducationPart/Form/EducationPartFormBuilder.php [hunk 12]
```diff
@@ -0,0 +1,66 @@
...
+     * @var array
+     */
+    protected $sections = [];
+
+    /**
+     * The form assets.
+     *
+     * @var array
+     */
+    protected $assets = [];
```
Comment (diashalabi) [inline addons/default/visiosoft/profile-module/src/EducationPart/Form/EducationPartFormBuilder.php:64]: unused properties

File: addons/default/visiosoft/profile-module/src/EducationPart/Table/EducationPartTableBuilder.php [hunk 13]
```diff
@@ -0,0 +1,61 @@
...
+     * @var array
+     */
+    protected $options = [];
+
+    /**
+     * The table assets.
+     *
+     * @var array
+     */
+    protected $assets = [];
```
Comment (diashalabi) [inline addons/default/visiosoft/profile-module/src/EducationPart/Table/EducationPartTableBuilder.php:59]: usused properties

File: addons/default/visiosoft/profile-module/src/ProfileModule.php [hunk 14]
```diff
@@ -30,6 +30,16 @@ class ProfileModule extends Module
...
             ],
         ],
+	    'education' => [
+	    	'buttons' => [
+	    		'new_education',
+		    ]
+	    ],
+	    'education_part' => [
+		    'buttons' => [
+			    'new_education',
```
Comment (diashalabi) [inline addons/default/visiosoft/profile-module/src/ProfileModule.php:40]: shouldn't be "new_education_part" instead?

File: addons/default/visiosoft/profile-module/src/ProfileModuleServiceProvider.php [hunk 15]
```diff
@@ -139,6 +143,8 @@ class ProfileModuleServiceProvider extends AddonServiceProvider
 
         // CacheController
         'ajax/get-user-info' => 'Visiosoft\ProfileModule\Http\Controller\CacheController@getUserInfo',
+	    'ajax/setEducation' => 'Visiosoft\ProfileModule\Http\Controller\MyProfileController@setEducation',
+	    'ajax/getEducation' => 'Visiosoft\ProfileModule\Http\Controller\MyProfileController@getEducation',
```
Comment (diashalabi) [inline addons/default/visiosoft/profile-module/src/ProfileModuleServiceProvider.php:147]: you can put these routes under // MyProfileController, and replace "ajax" with "api"

File: addons/default/visiosoft/profile-module/src/Seed/UsersFieldsSeeder.php [hunk 16]
```diff
@@ -124,21 +124,31 @@ public function run(
...
+	        [
+		        'name' => trans('visiosoft.module.profile::field.education_part.name'),
+		        'slug' => 'education_part',
+		        'type' => 'anomaly.field_type.select',
+	        ],
+	        [
+		        'name' => trans('visiosoft.module.profile::field.education_part_option.name'),
+		        'slug' => 'education_part_option',
+		        'type' => 'anomaly.field_type.select',
+	        ],
```
Comment (diashalabi) [inline addons/default/visiosoft/profile-module/src/Seed/UsersFieldsSeeder.php:151]: are we using this field "education_part_option"? or any of these fields in that matter? can you recheck the fields so we don't have unused fields

File: addons/default/visiosoft/profile-module/src/Profile/Profile/ProfileFormBuilder.php [hunk 17]
```diff
@@ -13,8 +13,13 @@ class ProfileFormBuilder extends FormBuilder
...
         'land_phone',
         'identification_number',
-        'education',
-        'state_of_education',
+        'education' => [
+	        'type' => 'anomaly.field_type.select',
+	        'config' => [
+		        'handler' => 'Visiosoft\ProfileModule\OptionHandler\EducationOptions@handle',
+	        ]
+        ],
```
Comment (diashalabi) [inline addons/default/visiosoft/profile-module/src/Profile/Profile/ProfileFormBuilder.php:21]: why change the type?

File: addons/default/visiosoft/profile-module/src/Http/Controller/MyProfileController.php [hunk 18]
```diff
@@ -246,4 +249,24 @@ public function updateAjaxProfile(UserRepositoryInterface $user)
         return response()->json(['status' => 'success', 'data' => $profile]);
     }
 
+	public function getEducation(Request $request)
+	{
+		$user = $this->userRepository->find(auth()->user()->getAuthIdentifier());
```
Comment (diashalabi) [inline addons/default/visiosoft/profile-module/src/Http/Controller/MyProfileController.php:254]: why not just use auth()->user()?

File: addons/default/visiosoft/profile-module/src/Http/Controller/MyProfileController.php [hunk 19]
```diff
@@ -246,4 +249,24 @@ public function updateAjaxProfile(UserRepositoryInterface $user)
...
+		$education = EducationModel::all();
+		$educationPart = EducationPartModel::query()->where('education_id', $user->education)->get();
+		return response()->json(['user' => $user, 'education' => $education, 'education-part' => $educationPart], 200);
+	}
+
+	public function setEducation(Request $request)
+	{
+		$user_id = auth()->user()->getAuthIdentifier();
```
Comment (diashalabi) [inline addons/default/visiosoft/profile-module/src/Http/Controller/MyProfileController.php:262]: why not just use auth()->user()?

**Review by diashalabi — CHANGES_REQUESTED** [review]

File: addons/default/visiosoft/profile-module/src/Http/Controller/MyProfileController.php (hunk 19 above)
Comment (muammertop) [reply addons/default/visiosoft/profile-module/src/Http/Controller/MyProfileController.php:262]: just need id

resim

File: addons/default/visiosoft/profile-module/src/Profile/Profile/ProfileFormBuilder.php (hunk 17 above)
Comment (muammertop) [reply addons/default/visiosoft/profile-module/src/Profile/Profile/ProfileFormBuilder.php:21]: relationship to select type

File: addons/default/visiosoft/profile-module/resources/assets/js/education.js (hunk 2 above)
Comment (muammertop) [reply addons/default/visiosoft/profile-module/resources/assets/js/education.js:22]: resim

Comment (diashalabi) [reply addons/default/visiosoft/profile-module/resources/assets/js/education.js:22]: `'<option value="">'+ choose_an_option +'</option>' or else the value will be "choose an option"`

File: addons/default/visiosoft/profile-module/resources/assets/js/education.js (hunk 3 above)
Comment (diashalabi) [reply addons/default/visiosoft/profile-module/resources/assets/js/education.js:40]: you can save on the backend

**Review by diashalabi — APPROVED** [review]

**Review by spektra2147 — APPROVED** [review]

**Pull request merged**